lib/public/AppFramework/AuthPublicShareController.php

```diff
@@ -84,6 +84,29 @@ protected function showAuthFailed(): TemplateResponse {
 		return new TemplateResponse('core', 'publicshareauth', ['wrongpw' => true], 'guest');
 	}
 
+	/**
+	 * The template to show after user identification
+	 *
+	 * @since 24.0.0
+	 */
+	protected function showIdentificationResult(bool $success): TemplateResponse {
+		return new TemplateResponse('core', 'publicshareauth', ['identityOk' => $success], 'guest');
+	}
+
+	/**
+	 * Validates that the provided identity is allowed to receive a temporary password
+	 *
+	 * @since 24.0.0
+	 */
+	abstract protected function validateIdentity(string $identityToken): bool;
+
+	/**
+	 * Generates a password
+	 *
+	 * @since 14.0.0
```

Comment:
```suggestion
	 * @since 24.0.0
```

Should this event be part of the public API?